src/components/Dialog/Dialog.tsx

```diff
@@ -261,8 +261,10 @@ export default class Dialog extends Component<DialogProps, DialogState> {
   stringifyJsonFields = (asset: AssetProps) => {
     const replaceNullWithEmptyString = (_: any, value: any) => (value === null) ? "" : value;
     asset.attributes.custom_fields = JSON.stringify(asset.attributes.custom_fields, replaceNullWithEmptyString);
-    asset.attributes.colors.dominant_colors = JSON.stringify(asset.attributes.colors.dominant_colors, replaceNullWithEmptyString);
     asset.attributes.tags = JSON.stringify(asset.attributes.tags, replaceNullWithEmptyString);
+    if (asset.attributes.colors?.dominant_colors) {
+      asset.attributes.colors.dominant_colors = JSON.stringify(asset.attributes.colors?.dominant_colors, replaceNullWithEmptyString);
+    }
```

Comment:
I think as a hotfix this could work. However, I'd feel a lot more confident if we could more strictly type the `asset` [prop](https://github.com/imgix/contentful/blob/9879ce69a82d5ed72f6604af4a68881e92ebf8e5/src/components/Dialog/Dialog.tsx#L28).

Are there response objects we can reference for each account tier that would allow us to delineate each key and whether or not it's optional? That way we'd catch these types of errors at build-time rather than run-time.

How do feel about that @sherwinski ?

Reply:
I definitely agree we should add stronger types for `asset` here, and potentially do the same for `source` as well. Looking at an example response from the lowest-tier we get an idea of which fields we can always expect values for:

```
    {
      "attributes": {
        "analyzed_content_warnings": false,
        "analyzed_faces": false,
        "analyzed_tags": false,
        "categories": [],
        "color_model": null,
        "color_profile": null,
        "colors": null,
        "content_type": "image/jpeg",
        "custom_fields": null,
        "date_created": 1643753719,
        "date_modified": 1643753719,
        "description": null,
        "dpi_height": null,
        "dpi_width": null,
        "face_count": null,
        "file_size": 902247,
        "has_frames": false,
        "media_height": null,
        "media_kind": "UNKNOWN",
        "media_width": null,
        "name": null,
        "origin_path": "/delete/squid-games.jpg",
        "source_id": "61f9ad2bb4859c6512b01c71",
        "tags": null,
        "uploaded_by": "5be48f6b3f876f0001cd2198",
        "uploaded_by_api": false,
        "warning_adult": null,
        "warning_medical": null,
        "warning_racy": null,
        "warning_spoof": null,
        "warning_violence": null
      },
      "id": "61f9ad2bb4859c6512b01c71/delete/squid-games.jpg",
      "type": "assets"
    },
```

Because of the user impact from rolling back the latest changes, I think we should move forward with the hotfix for now and revisit this in the near future.

> Are there response objects we can reference for each account tier that would allow us to delineate each key and whether or not it's optional? That way we'd catch these types of errors at build-time rather than run-time.

In the process of figuring this out now 👍